vine/src/compile/build_stages.rs

```diff
@@ -132,7 +132,11 @@ impl Compiler<'_> {
       ExprKind::Loop(body) => self.lower_loop(body),
       ExprKind::While(cond, body) => self.lower_while(cond, body),
       ExprKind::If(cond, then, els) => self.lower_if(cond, then, els),
-      ExprKind::Return(r) => self.lower_return(r),
+      ExprKind::Return(None) => {
+        let unit = Expr { span: Span::NONE, kind: ExprKind::Tuple(vec![]) };
+        self.lower_return(&unit)
```

Comment:
I think it would be cleaner for `lower_return` to take an `Option<Expr>` and just not write to the local if it's `None`

Reply:
woah didn't expect that to work. Right now is it undefined behaviour to assign the result of function that returns nothing to a var?

Reply:
no, it's just that uninitialized variables are represented with an eraser node, which is the same as an empty tuple

Reply:
or, one can think of it as every variable being initialized as an empty tuple